fdbclient/NativeAPI.actor.cpp

```diff
@@ -3351,7 +3351,7 @@ void Transaction::setOption( FDBTransactionOptions::Option option, Optional<Stri
 			if (info.debugID.present()) {
 				TraceEvent(SevInfo, "TransactionBeingTraced")
 					.detail("DebugTransactionID", trLogInfo->identifier)
-					.detail("ServerTraceID", info.debugID.get().toString());
+					.detail("ServerTraceID", info.debugID.get().first());
```

Comment:
I think `.shortString()` is the preferred way to do this.

Reply:
For better or worse, `first()` was used in every `TraceEvent` I was able to find by (rip)grepping. In theory, `TraceEvent` could (and arguably) should avoid allocating a temporary string for the integer. Using `shortString()` would allocate a temporary even if we fixed `TraceEvent`.

Reply:
Do you have an example? The only place I know of using `first()` off the top of my head was for the trace batch, which effectively results in the value being transformed in the same was as with `shortString()`.

Reply:
https://gist.github.com/Daniel-B-Smith/bb85ee126d4302a64ea9c015f2441ac8

Reply:
Yeah, those all look like the `traceBatch`, which takes as an argument a uint64_t but converts it to a string the same way that `shortString` does. If you are passing the value into a detail, I think `shortString` is going to do what you want (print the int as a hexadecimal number). I think, though, that most of the time we put a server ID in a trace event detail, we do actually use the long form. Because this is used in tandem with the transaction tracing, it seems reasonable to use the `shortString` that matches what is used in the trace batches.